lib/_http_client.js

```diff
@@ -164,22 +164,22 @@ function ClientRequest(options, cb) {
     this.once('response', cb);
   }
 
-  var headersArray = Array.isArray(options.headers);
+  let headersArray = Array.isArray(options.headers);
   if (!headersArray) {
     if (options.headers) {
-      var keys = Object.keys(options.headers);
-      for (var i = 0; i < keys.length; i++) {
-        var key = keys[i];
+      let keys = Object.keys(options.headers);
+      for (let i = 0; i < keys.length; i++) {
```

Comment:
Performance-wise V8 was supposed to have fixed this particular use case awhile ago, but you never know if there are weird edge cases or something yet. That is the main reason for my comment about checking for performance regressions just to be sure.

Reply:
I believe that @bmeurer flagged this as still being potentially problematic (in some limited scenarios) in an earlier PR that tried to get rid of the lint rule. Either way, this wouldn't pass the CI (linter) so it will need to be changed.